app/utils/html-utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ *
+ * @param {String} htmlString String containing valid HTML
+ * @param {String} id ID of the element whose content needs to be replaced
+ * @param {String} newValue New value of the element
+ */
+export function replaceById(htmlString, id, newValue) {
+  const template = document.createElement('template');
+  htmlString = htmlString.trim();
+  template.innerHTML = htmlString;
+  const element = template.content.querySelector(`#${id}`);
+  if (element) {
+    element.innerHTML = newValue;
+    return template.innerHTML;
+  } else {
+    console.debug(
+      `No element with id "${id}" found, could not set innerHTML`
+    );
```

Comment:
This was intended for a developer, I was foreseeing this function being used in other circumstances as well. I guess raising an error might be a better approach so it can be handled by the caller.

Reply:
Update: I'll just remove it, the caller can check if the substring is present beforehand if it needs to know